src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,496 @@
+
+  /** add channel to frequency */
+  if (channel_encoded <= 3)
+    frequency += channel_encoded * 5;
+  else if (channel_encoded <= 7)
+    frequency += channel_encoded * 5 + 5;
+  else if (channel_encoded <= 11)
+    frequency += channel_encoded * 5 + 10;
+  else if (channel_encoded <= 15)
+    frequency += channel_encoded * 5 + 15;
+
+  /** generate RadioFrequency object */
+  RadioFrequency radio_frequency = RadioFrequency();
+  radio_frequency.SetKiloHertz(frequency);
+
+  return radio_frequency;
+}
+
+uint16_t AR62xxDevice::FrequencyToAR62xxId(const RadioFrequency freq)
+{
+  /** get frequency in kiloherz */
+  unsigned frequency = freq.GetKiloHertz();
+
+  /** check if frequency is not in range */
+  if (frequency < MIN_FREQUENCY || frequency > MAX_FREQUENCY)
+    return 0;
+
+  /** get offset of frequency to the lowest frequency which is possible */
+  unsigned frequency_offset = frequency - MIN_FREQUENCY;
+
+  /** generate frequency-id */
+  uint16_t frequency_id = std::round(frequency_offset * MAX_FREQUENCY_ID / FREQUENCY_RANGE);
+
+  /** append frequency bitmask as channel has to be added separately */
+  frequency_id &= FREQUENCY_BITMASK;
+
+  /** get channel out of frequency */
+  uint8_t channel = frequency % 100;
+
+  /** encode channel and add to frequency */
+  if (channel <= 15)
+    frequency_id += channel / 5;
+  else if (channel <= 40)
+    frequency_id += channel / 5 - 1;
+  else if (channel <= 65)
+    frequency_id += channel / 5 - 2;
+  else if (channel <= 90)
+    frequency_id += channel / 5 - 3;
+
+  return (frequency_id);
+}
+
+static uint16_t CRCBitwise(uint8_t *data, size_t len)
+{
+  uint16_t crc = 0x0000;
+  for (size_t j = len; j > 0; j--)
+  {
+    crc ^= (uint16_t)(*data++) << 8;
+    for (int i = 0; i < 8; i++)
+    {
+      if (crc & 0x8000)
+        crc = (crc << 1) ^ 0x8005;
+      else
+        crc <<= 1;
+    }
+  }
+  return (crc);
+}
+
+unsigned int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, unsigned int active_passive, const RadioFrequency frequency)
+{
+  /** length of the command */
+  unsigned int command_length = 0;
+
+  /** convert the active-frequency to an AR62xx-ID */
+  uint16_t active_frequency_idx = FrequencyToAR62xxId(active_frequency);
+
+  /** convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /** command-start */
+  command[command_length++] = 0xA5;
+
+  /** add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /** set active station */
+  if (active_passive == ACTIVE_STATION)
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** set passive station */
+  else
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** setting frequencies */
+  command[command_length++] = 22;
+  command[command_length++] = GetNthByte(active_frequency_idx, 1);
+  command[command_length++] = GetNthByte(active_frequency_idx, 0);
```

Comment:
This is a very obscure way of serializing a 16 bit integer as little-endian! I don't like the whole `GetNthByte()` function, because it's part of the obscurity.

Reply:
what do you expect to be done?

Reply:
replaced the GetNthByte